pkgs/by-name/as/asm-lsp/package.nix

```diff
@@ -18,13 +22,16 @@ rustPlatform.buildRustPackage {
     hash = "sha256-0GB3tXZuCu3syh+RG+eXoliZVHMPOhYC3RchSSx4u5w=";
   };
 
-  nativeBuildInputs = [
-    pkg-config
-  ];
+  nativeBuildInputs =
+    [
+      pkg-config
+    ]
+    ++ lib.optionals stdenv.buildPlatform.isDarwin [
```

Comment:
Dependencies such as libraries/frameworks need to be added to `buildInputs`. You can find more information [here](https://nixos.org/manual/nixpkgs/stable/#ssec-stdenv-dependencies-reference).